packages/gator-permissions-controller/src/decodePermission/decodePermission.ts

```
@@ -223,6 +259,56 @@ export const getPermissionDataAndExpiry = ({
   return { expiry, data };
 };
 
+/**
+ * Extracts the expiry timestamp from a delegation permission context.
+ *
+ * This function decodes the delegation context, locates the TimestampEnforcer
+ * caveat, and extracts the expiry timestamp (timestampBeforeThreshold).
+ *
+ * @param permissionContext - The delegation context as a hex string
+ * @param chainId - The chain ID as a hex string
+ * @returns The expiry timestamp in seconds, or null if no expiry exists or on error
+ */
+export const extractExpiryFromPermissionContext = (
+  permissionContext: Hex,
+  chainId: Hex,
+): number | null => {
+  try {
+    const delegations = decodeDelegations(permissionContext);
+
+    if (delegations.length !== 1) {
+      return null;
+    }
+
+    const delegation = delegations[0];
+    if (!delegation) {
+      return null;
+    }
+
+    const chainIdNumber = hexToNumber(chainId);
+    const contractsByChain = DELEGATOR_CONTRACTS[DELEGATION_FRAMEWORK_VERSION];
+    const contracts = contractsByChain[chainIdNumber];
+
+    if (!contracts) {
+      return null;
+    }
+
+    const { timestampEnforcer } = getChecksumEnforcersByChainId(contracts);
+
+    const timestampCaveat = delegation.caveats.find(
+      (caveat) => getChecksumAddress(caveat.enforcer) === timestampEnforcer,
+    );
+
+    if (!timestampCaveat) {
+      return null;
+    }
+
+    return extractExpiryFromCaveatTerms(timestampCaveat.terms);
+  } catch {
+    return null;
```

Is there a reason we're silencing the error here? It might be better to log the error before return null;

---

Added the log